src/corefx/System.Globalization.Native/calendarData.cpp

```diff
@@ -147,6 +151,84 @@ const char* GetCalendarName(CalendarId calendarId)
 
 /*
 Function:
+GetCalendarName
+
+Gets the associated CalendarId for the ICU calendar name.
+*/
+CalendarId GetCalendarId(const char* calendarName)
+{
+	if (strcmp(calendarName, "gregorian") == 0)
+		//TODO: what about the other gregorian types?
+		return GREGORIAN;
+	else if (strcmp(calendarName, "japanese") == 0)
+		return JAPAN;
+	else if (strcmp(calendarName, "buddhist") == 0)
+		return THAI;
+	else if (strcmp(calendarName, "hebrew") == 0)
+		return HEBREW;
+	else if (strcmp(calendarName, "chinese") == 0)
+		//TODO: what about these other lunisolar types?
+		//JAPANESELUNISOLAR:
+		//TAIWANLUNISOLAR:
+		//KOREANLUNISOLAR:
+		return CHINESELUNISOLAR;
+	else if (strcmp(calendarName, "dangi") == 0)
+		return KOREA;
+	else if (strcmp(calendarName, "persian") == 0)
+		return PERSIAN;
+	else if (strcmp(calendarName, "islamic") == 0)
+		return HIJRI;
+	else if (strcmp(calendarName, "islamic-umalqura") == 0)
+		return UMALQURA;
+	else if (strcmp(calendarName, "roc") == 0)
+		return TAIWAN;
+	else
+		return UNINITIALIZED_VALUE;
+}
+
+/*
+Function:
+GetCalendars
+
+Returns the list of CalendarIds that are available for the specified locale.
+*/
+extern "C" int32_t GetCalendars(const UChar* localeName, CalendarId* calendars, int32_t calendarsCapacity)
+{
+	Locale locale = GetLocale(localeName);
+	if (locale.isBogus())
+		return 0;
+
+	UErrorCode err = U_ZERO_ERROR;
+	LocalPointer<StringEnumeration> stringEnumerator(Calendar::getKeywordValuesForLocale("calendar", locale, TRUE, err));
+
+	if (stringEnumerator.isNull() || U_FAILURE(err))
+		return 0;
+
+	int stringEnumeratorCount = stringEnumerator->count(err);
+	if (U_FAILURE(err))
+		return 0;
+
+	int calendarsReturned = 0;
+	for (int i = 0; i < stringEnumeratorCount && calendarsReturned < calendarsCapacity; i++)
+	{
+		int32_t calendarNameLength = 0;
+		const char* calendarName = stringEnumerator->next(&calendarNameLength, err);
+		if (U_SUCCESS(err))
+		{
+			CalendarId calendarId = GetCalendarId(calendarName);
+			if (calendarId != UNINITIALIZED_VALUE)
+			{
+				calendars[calendarsReturned] = calendarId;
+				calendarsReturned++;
+			}
+		}
+	}
+
+	return calendarsReturned;
```

Comment:
I will do this in the managed code, to keep the native code as simple as possible.